test/src/lib/deck.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const assert = require('assert');
+const expect = require('chai').expect;
+
+const build = require('../../../src/lib/deck-builder');
+const Card = require('../../../src/lib/card');
+const Deck = require('../../../src/lib/deck');
+
+
+describe('Deck', function() {
+    let deck;
+
+    beforeEach(function() {
+        deck = new Deck(build());
+    });
+
+    context('draw', function() {
+        it('should return one Card in an array', function() {
+            expect(deck.draw()).to.be.an('array').that.has.lengthOf(1);
+        });
+
+        it('should be an instance of a Card', function() {
+            expect(deck.draw()[0]).to.be.instanceOf(Card);
+        });
+    });
+
+    context('draw 10', function() {
+        it('should return 10 Cards in an array', function() {
+            expect(deck.draw(10)).to.be.an('array').that.has.lengthOf(10);
+        });
+    });
+
+    context('shuffle', function() {
+        it('should rearrange the Cards in the Deck', function() {
+            const staticDeck = new Deck(build());
```

Comment:
Instead of instantiating two decks, make a deep clone of the original deck, then compare the shuffled version against the original version.